### PR TITLE
GH-37073: [Java] JDBC: Only use username/pass auth if token is not provided

### LINKED ISSUE #37073: [Java][JDBC-driver] Ignore username/password auth if they are empty and a `token` is provided
### Describe the enhancement requested

We've run into some issues with our service where users are using tools that always provide a username/password even if none are provided and a token is passed in as part of the connection params. This causes the driver to go through the Handshake flow using username/password which fails since we don't support that.

Ideally, tools should not attempt to use username/password if none are provided, but it's impossible to force tools to change their implementation. Instead, we could check if the username/password are empty and the `token` is provided in the connection parameters and choose `token` auth instead.

If this seems reasonable I'd be happy to put up a PR for it.

### Component(s)

Java

### PR DESCRIPTION
**Pull request by aiguofer**
### Are these changes tested?

Yes 

### Are there any user-facing changes?

Provided token authentication will now take precedence over username/password.
* Closes: #37073

### CODE REVIEW

File: java/flight/flight-sql-jdbc-core/src/test/java/org/apache/arrow/driver/jdbc/ConnectionTest.java [hunk 1]
```diff
@@ -102,6 +102,33 @@ public void testUnencryptedConnectionShouldOpenSuccessfullyWhenProvidedValidCred
     }
   }
 
+  /**
+   * Checks if a token is provided it takes precedence over username/pass. In this case,
+   * the connection should fail if a token is passed in.
+   *
+   * @throws SQLException on error.
+   */
+  @Test(expected = SQLException.class)
```
Comment (lidavidm) [inline java/flight/flight-sql-jdbc-core/src/test/java/org/apache/arrow/driver/jdbc/ConnectionTest.java:111]: nit: could we use assertThrows to make it clearer where we expect to throw (and possibly validate the error message too)?

**Review by lidavidm — APPROVED** [review]
Thanks!

File: java/flight/flight-sql-jdbc-core/src/test/java/org/apache/arrow/driver/jdbc/ConnectionTest.java (hunk 1 above)
Comment (aiguofer) [reply java/flight/flight-sql-jdbc-core/src/test/java/org/apache/arrow/driver/jdbc/ConnectionTest.java:111]: sure thing!

**Review by lidavidm — APPROVED** [review]
Thanks!

**Pull request merged**